lib/background/payment_hash_poller.dart

```diff
@@ -34,22 +35,17 @@ class PaymentHashPoller {
     final injector = ServiceInjector();
     final breezLib = injector.breezSDK;
     try {
-      final ListPaymentsRequest req = ListPaymentsRequest(
-        filters: [PaymentTypeFilter.Received],
-        fromTimestamp: DateTime.now().subtract(const Duration(minutes: 30)).millisecondsSinceEpoch,
-      );
-      final List<Payment> paymentList = await breezLib.listPayments(req: req);
-      for (var payment in paymentList) {
-        final detailsData = payment.details.data;
-        final isPaymentReceived = payment.status == PaymentStatus.Complete &&
-            detailsData is LnPaymentDetails &&
-            detailsData.paymentHash == paymentHash;
-        if (isPaymentReceived) {
+      final Payment? payment = await breezLib.paymentByHash(hash: paymentHash);
+      if (payment != null) {
+        if (payment.status == PaymentStatus.Complete) {
           print("Payment received! Stop polling.");
           timer.cancel();
           print("Completed payment_received task for $paymentHash.");
           taskCompleter.complete(true);
-          return;
+          
+          // TODO Remove this delay when we have a reliable way tell if the signer is connected (Greenlight webhooks)
+          // Artificial delay to keep the signer running for longer, to ensure the preimage is sent
+          return await Future.delayed(const Duration(seconds: 10));
```

Comment:
For the delay to take effect perhaps we should add it before we complete the task?